### PR TITLE
build: update litd to version `v0.12.1-alpha` 

### PR DESCRIPTION
**Pull request by Roasbeef**

### CODE REVIEW

**Comment by ViktorTigerstrom** [comment]
Updated the PR to fix the failing itests. Also bumped the loop & taproot-assets dependencies! Finally I also edited the `README` file.

**Comment by ViktorTigerstrom** [comment]
Updated the itests to be compatible with the updates of taproot-assets `v0.3.1-alpha`.

**Review by ellemouton — APPROVED** [review]
lgtm!

File: itest/litd_mode_integrated_test.go [hunk 1]
```diff
@@ -319,7 +319,7 @@ var (
 		name:              "taprpc-whitelist",
 		macaroonFn:        emptyMacaroonFn,
 		requestFn:         tapUniverseRequestFn,
-		successPattern:    "\"num_assets\":",
+		successPattern:    "\"runtime_id\":",
```
[the remark this answers is not in the document]
Comment (ViktorTigerstrom) [reply itest/litd_mode_integrated_test.go:322]: Ah yeah, good catch! squashed the commits :)

**Review by jamaljsr — APPROVED** [review]
tACK LGTM 🚀

Tested Loop & Pool via the UI. All good.

**Pull request merged**